### ISSUE TITLE
goto-line command

### ISSUE THREAD

**Issue by xcombelle** [howl-editor/howl#122]
I realized this goto-line command. feel free to integrate with current code
If you prefer I realize a proper pull request says so.

```moonscript
-- code licensed under same license than howl (MIT)

import app, command,interact from howl

command.register
  name: 'goto-line'
  description: 'Goto specified line'
  input: () ->
    line_str = interact.read_text prompt:""
    return tonumber(line_str)

  handler: (line_no) ->
    editor = app.editor
    if line_no < 1
      line_no = 1
    if line_no > #editor.buffer.lines
      line_no = #editor.buffer.lines
    editor.cursor.line = line_no
    editor.cursor.column_index = 1
    editor.line_at_top = line_no

howl.bindings.push {
  editor: {
      alt_g: 'goto-line'
  }
}
```

by the way it is very nice software (and not too hard to customize for its pre 1.0 status)

**Comment by kirbyfan64** [comment]
I had suggested this feature in https://github.com/howl-editor/howl/issues/112#issuecomment-148055551, except bound to `ctrl_g`.

**Comment by nilnor** [comment]
It seem there's a pent-up need for a goto line command, so let's add it :) I actually use the same thing quite often, but as I use the VI emulation it's already built in there, so I do see a need for it. 

@kirbyfan64: `ctrl_g` is already bound to `buffer-grep`, so something other than will be needed unless we want to break the current mapping, but at least I am well used to `ctrl_g` meaning grep by now.

@xcombelle if you want to open a PR, then please go ahead! Some initial feedback:

- The error checking for line number is nice, but is already handled deeper down and so not necessary
-  There's a new (not yet documented) method `move_to` method for the cursor now, so the moving could be done as `cursor\move_to line: line_no, column: 1`
- I personally don't think that always scrolling so that the specified line as at the top makes sense - if it's in view then I prefer the cursor to just go there. In the future we might flash the line to indicate where it is. Others will have to chime in here though.

Finally, nice to hear that you had a good experience customizing it. If there's anything in particular you think could make it easier still then let us know :)

**Comment by shalabhc** [comment]
As I mentioned in #112 I use `buffer-grep` as a 'goto line' command (with preview!) and it works quite well.. However, a dedicated command for this makes sense.

@kirbyfan64 - `ctrl_g` for `buffer-grep` is one of my most used shortcuts and I would prefer to not change that binding.

@nilnor:
> I personally don't think that always scrolling so that the specified line as at the top makes sense - if it's in view then I prefer the cursor to just go there. In the future we might flash the line to indicate where it is. Others will have to chime in here though.

I also feel that if the line is in view it probably shouldn't scroll. If the line is not in view, my personal preference would be `line_at_middle` rather than `line_at_top`.

I don't think flashing is required since there is the current line highlight, but I'm not against it. While we can't flash, an additional transient highlight should be easy to do with something like this (untested):

```moonscript
line = buffer.lines[line_nr]
howl.ui.highlight 'search', buffer, line.start_pos, line.size
```

**Comment by kirbyfan64** [comment]
Whoops, I forgot that `ctrl_g` was already taken.

I would prefer no flashing and a simple scroll to the proper line (e.g. nothing like `line_at_middle`), if not just for consistency with other commands.

**Comment by nilnor** [comment]
@xcombelle Did you want to open a PR for this, or should someone else do it?

**Comment by nilnor** [comment]
Fixed via #131.
